actions/kpops-runner/action.yaml

```diff
@@ -0,0 +1,47 @@
+name: "Kpops runner"
+description: "Run Kpops command line with the given params"
+
+inputs:
+  mode:
+    description: "The command mode you want to use to launch kpops"
+    required: true
+  pipeline:
+    description: "Pipeline file you want to run"
+    required: true
+  working-directory:
+    description: "The root directory containing the config.yaml, pipelines folder and defaults"
+    default: "."
+  execute:
+    description: "Set execution flag"
+    default: "false"
+  defaults:
+    description: "Defaults folder path"
+    default: "defaults"
+  config:
+    description: "Config.yaml file path"
+    default: "config.yaml"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Check out repository code
+      uses: actions/checkout@v3
+
+    - name: Install Kpops
+      shell: bash
+      run: pip install kpops==0.2.0.dev20221122085523 --extra-index-url https://test.pypi.org/simple/
+      
+    
+    - name: Dry-run ${{ inputs.pipeline }} pipeline
+      shell: bash
+      working-directory: ${{inputs.working-directory}}
+      run: |
+         kpops generate ${{ inputs.pipeline }} --defaults ${{ inputs.defaults }} --config ${{ inputs.config }}
+         kpops ${{inputs.mode}} ${{ inputs.pipeline }} --defaults ${{ inputs.defaults }} --config ${{ inputs.config }}
```

Comment:
I don't think you need to use `generate` here. You can directly use:

```
kpops ${{inputs.mode}} ${{ inputs.pipeline }} --defaults ${{ inputs.defaults }} --config ${{ inputs.config }} --dry-run
```

Reply:
you mean giving generate using `inputs.mode` ?

Reply:
Generate should be a separate step that always runs as it is more for debugging purposes and doesn't do anything

Reply:
Should I let this as it is or should I change something? I didn't understand @philipp94831

Reply:
You did it right